ft-sdk/src/layout.rs

```diff
@@ -11,10 +11,15 @@ pub trait Action<L, E>
 where
     E: std::fmt::Debug + From<ft_sdk::Error>,
 {
-    fn validate(c: &mut L) -> Result<Self, E>
+    fn validate(c: &mut L, start_time: chrono::DateTime<chrono::Utc>) -> Result<Self, E>
     where
         Self: Sized;
-    fn action(&self, c: &mut L) -> Result<ActionOutput, E>
+
+    fn action(&self, c: &mut L, start_time: chrono::DateTime<chrono::Utc>) -> Result<ActionOutput, E>
+    where
+        Self: Sized;
+
+    fn action_(&self, c: &mut L, start_time: chrono::DateTime<chrono::Utc>) -> Result<i64, E>
```

Comment:
Nopes. Why do we need this in our trait? If you can to create other methods, create it, it does not have to be added to trait. Add another impl block.